fix : Adjust file Tab option buttons for small screen

yadavshubham01 opened the pull request:
## Proposed Changes

- Fixes #12455 
-  add width for "Generate Discharge Summary" button to span the full width on small devices 
-  alignment of file tab list with search field and others button like filter ,etc; [#comment](https://github.com/ohcnetwork/care_fe/issues/12455#issuecomment-2920340211)

## Screenshot 

![Screenshot 2025-06-10 211538](https://github.com/user-attachments/assets/9b5b1b20-2b59-4aaf-9891-c0231b887542)

![image](https://github.com/user-attachments/assets/60b58cee-0683-4639-9989-d0a09bacf336)


@ohcnetwork/care-fe-code-reviewers

## Merge Checklist

- [ ] Add specs that demonstrate bug / test a new feature.
- [ ] Update [product documentation](https://docs.ohc.network).
- [ ] Ensure that UI text is kept in I18n files.
- [x] Prep screenshot or demo video for changelog entry, and attach it to issue.
- [x] Request for Peer Reviews
- [x] Completion of QA in Mobile Devices
- [ ] Completion of QA in Desktop Devices


<!-- This is an auto-generated comment: release notes by coderabbit.ai -->
## Summary by CodeRabbit

- **Style**
  - Improved responsive spacing and layout for search input containers, filter buttons, and file upload controls across multiple tabs.
  - Enhanced tab list and tab trigger styling for better alignment and adaptability across different screen sizes.
  - Updated button and icon sizing to ensure consistent appearance on smaller screens.
<!-- end of auto-generated comment: release notes by coderabbit.ai -->

yadavshubham01 committed:
make buttons reponsive for mobile screen size

AdityaJ2305 commented on src/components/Files/FilesTab.tsx:
```diff
@@ -67,24 +69,29 @@ export const FilesTab = (props: FilesTabsProps) => {
-        <TabsList className={type != "encounter" ? "mt-2" : ""}>
+        <TabsList
+          className={cn(
+            type != "encounter" ? "mt-2" : "",
```
Should use `&&` here

yadavshubham01 committed:
fixing the condition

Jacobjeevan committed:
adjust flex containers

AdityaJ2305 commented on src/components/Files/DischargeSummarySubTab.tsx:
```diff
@@ -271,7 +271,7 @@ export const DischargeTab = ({
     return (
       <DropdownMenu>
-        <DropdownMenuTrigger asChild>
+        <DropdownMenuTrigger asChild className="w-full sm:w-auto">
```
Shouldn't we have className styling to the button (child comp) itself ? 🤔

yadavshubham01 replied:
yeah , we can add this styling to child button itself and its working .

AdityaJ2305 commented on src/components/Files/DischargeSummarySubTab.tsx:
```diff
@@ -326,7 +326,7 @@ export const DischargeTab = ({
     return (
       <DropdownMenu>
-        <DropdownMenuTrigger asChild>
+        <DropdownMenuTrigger asChild className="w-full sm:w-auto">
```
same here

AdityaJ2305 commented on src/components/Files/FileSubTab.tsx:
```diff
@@ -284,7 +284,7 @@ export const FilesPage = ({
     return (
       <DropdownMenu>
-        <DropdownMenuTrigger asChild>
+        <DropdownMenuTrigger asChild className="w-full sm:w-auto">
```
same

AdityaJ2305 commented on src/components/Files/FileSubTab.tsx:
```diff
@@ -345,7 +345,7 @@ export const FilesPage = ({
     return (
       <DropdownMenu>
-        <DropdownMenuTrigger asChild>
+        <DropdownMenuTrigger asChild className="w-full sm:w-auto">
```
same

AdityaJ2305 commented on src/components/Files/FileSubTab.tsx:
```diff
@@ -345,7 +345,7 @@ export const FilesPage = ({
     if (!canEdit) return <></>;
```
return null 

```suggestion
    if (!canEdit) return null;
```

yadavshubham01 committed:
add style to button

AdityaJ2305 commented on src/components/Files/FileSubTab.tsx:
```diff
@@ -342,13 +342,13 @@ export const FilesPage = ({
     return (
       <DropdownMenu>
-        <DropdownMenuTrigger asChild>
+        <DropdownMenuTrigger asChild className="">
```
```suggestion
        <DropdownMenuTrigger asChild>
```

yadavshubham01 committed:
remove classname

yadavshubham01 committed:
add grid style

yadavshubham01 committed:
Merge branch 'develop' into fix/tabList-button-mobileSize

yadavshubham01 committed:
Merge branch 'develop' into fix/tabList-button-mobileSize

yadavshubham01 committed:
resolve the conflict and fixing

yadavshubham01 committed:
Merge branch 'develop' into fix/tabList-button-mobileSize

yadavshubham01 committed:
Merge branch 'develop' into fix/tabList-button-mobileSize

yadavshubham01 committed:
fixing subFile-tab buttons

yadavshubham01 committed:
Merge branch 'develop' into fix/tabList-button-mobileSize

yadavshubham01 commented:
Hey @nihal467 ,
Just to update you the issue originally occurred in the Subfile tab, but since we've now removed the Discharge Summary button, that specific problem is no longer relevant.
However, apart from that, we also fixed several minor issues:
- alignment of file tab list with search field and others button like filter 
- Updated button and icon sizing to ensure consistent appearance on smaller screens.

<img width="630" height="891" alt="Screenshot 2025-07-23 143835" src="https://github.com/user-attachments/assets/098ad61c-77df-4c33-ab6b-f2258a418670" />
